### DISCUSSION TITLE
Possible to temporarily disable rebalancing?

### DISCUSSION THREAD

**Discussion by Blackpaw** [moosefs/moosefs#427] [Q&A]
Is it possible to temporarily disable regular and internal rebalancing?

I have a setup where I added new disks am migratinglarge amounts of data in place - as a result, the disk usage is massively in balanced and keeps changing as I delete old files while I migrate. The imbalance doesn't matter as I have a large amount of free space to accommodate it.

However I have internal rebalance loads per chunkserver going on and I presume rebalance between chunkservers - there is a lot of reads happening when nothing should be accessing the mfs mount, and I presume it is impacting my data migration as it is going very slowly, with writes cycling between 70 MB/s to a few hundred K/s.

I'd really like to just get the data transferred and then let the system rebalance to its heart's content.

**Comment by jkiebzak** [comment]
I've never tested this, but I think you may be able to change the values for `CHUNKS_WRITE_REP_LIMIT` within the `mfsmaster.cfg`. I'd try changing the values to `0,0,0,0`, then reload the master. Just don't forget to change it back (and reload).

**Comment by borkd** [comment]
I would not mess with replication limits, especially ones meant to protect the data. Start with bumping the acceptable difference to the max?

**Comment by chogata** [comment]
You can mess with limits if you know what you are doing ;)

Setting `ACCEPTABLE_PERCENTAGE_DIFFERENCE = 10` (which is the maximum allowed value) will help, but it will not stop all the replications, only "rebalance between other servers (very low or very high space usage)" and if the difference is bigger than max (10%), it will not work anyway.

Instead, you can change the replication limits, but first, not all of them to 0, just the third and the fourth, those are responsible for both "normal rebalance" and "high space difference rebalance", second, **both sets of limits**, not just the WRITE limits. Changing only WRITE will make the master try to do the replications (because reading can be done) but give up (because writing cannot be done), needlessly burdening the processor.
So try this (first two limits taken from MooseFS defaults, adjust if you changed them):
```
CHUNKS_WRITE_REP_LIMIT = 2,1,0,0
CHUNKS_READ_REP_LIMIT = 10,5,0,0
```
and it should help your case.
Don't forget to change them back later, when you want the actual rebalancing to start :)

**Reply by Blackpaw** [reply]
Thanks all, my data transfer finally finished late last night, so moot point now, but will keep in mind for any future efforts.

Have cleared off the old system now (lizardfs) and the pool has finished revalancing now, nice even distribution of data.
